src/libsystemd/sd-bus/bus-match.c

```diff
@@ -878,9 +878,9 @@ char *bus_match_to_string(struct bus_match_component *components, unsigned n_com
         }
 
         r = fflush_and_check(f);
+        safe_fclose(f);
         if (r < 0)
                 return NULL;
```

Comment:
Is it not necessary to free `buffer` on failure? How about the following?
```
_cleanup_free_ char *buffer = NULL;
...
return TAKE_PTR(buffer);
```

Reply:
I merged your PR and rebased my changes on top and used the same pattern here.